OptionSelector/DetailTableViewController.m

```diff
@@ -0,0 +1,109 @@
+//
+//  DetailTableViewController.m
+//  Glycemic Index of Common Foods OptionSelector App
+//
+//  Created by Ayuna Vogel on 8/10/15.
+//  Copyright (c) 2015 Ayuna Vogel. All rights reserved.
+//
+
+#import "DetailTableViewController.h"
+#import "CQCategory.h"
+#import "MainTableViewController.h"
+
+@interface DetailTableViewController ()
+
+@end
+
+@implementation DetailTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.navigationItem.title = @"🍅🍉Check to Track🍕🍔";
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    //the method belows deals with having a NSArray *options that is a property of class CQCategory and is an array that contains a NSDictionary and NSStrings
+    if (self.options != nil) {
+        return self.options.count;
+    }
+    return self.category.options.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ItemCellIdentifier" forIndexPath:indexPath];
+    
+    // object will either be an NSString or a NSDictionary
+    id object;
```

Comment:
Another way to organize this instead of having to use isKindOfClass would be to create a separate class that stores a string label and an array of children (or a dict of children) and has a method like isTerminalNode that returns true if there are no children. This abstracts the logic a bit so if you decide to change from array to dict later, you can do that more easily.

(This is a very common structure in programming called a [tree](https://en.wikipedia.org/wiki/Tree_(data_structure)), and there are many ways to implement them.)